### PR TITLE
Docs [#20]

### PR DESCRIPTION
**Pull request by k33rs**
Improved documentation following comments in issue [#20](https://github.com/martel-innovate/deep-log-inspection/issues/20).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,22 +6,36 @@ This repository contains resources for deploying the Deep Log Inspection system
...
+The compose file can be found [here](https://github.com/martel-innovate/deep-log-inspection/blob/master/log-server/docker-compose.yml).
+
+If needed, the environment can be completely cleaned:
+
+    docker stack rm deeplog
+
+For the full CLI reference, please refer to the [Docker CLI reference guide](https://docs.docker.com/engine/reference/commandline/cli/).
+
+## Services and DNS
+The system is meant to be deployed on a multi-node cluster, in which the master node is accessible with three valid domain names. Let `example.com` be the main domain name and `elastic.example.com` and `kibana.example.com` two alternative domains (in this page and throughout the full guide). Then, the following services should be exposed on the master node:
```
Comment (chicco785) [inline README.md]: `elastic.example.com` and `kibana.example.com` give that are mapped to an ip, are name servers (and eventually subdomains)

File: README.md [hunk 2]
```diff
@@ -6,22 +6,36 @@ This repository contains resources for deploying the Deep Log Inspection system
...
 
 * monasca-log-api at `example.com`
 * elasticsearch at `elastic.example.com`
```
Comment (chicco785) [inline README.md]: kibana should be listed or not?

File: README.md [hunk 3]
```diff
@@ -31,15 +45,13 @@ The system is meant to be deployed on a multi-node cluster, in which the master
...
 To bring keystone up along with the other services, deploy it to the same stack:
 
     docker stack deploy -c docker-compose-keystone.yml deeplog
 
-If needed, `docker stack rm deeplog` can be used to completely clean the environment between runs.
-
-For the full CLI reference, please refer to the [Docker CLI reference guide][6].
+The compose file for keystone can be found [here](https://github.com/martel-innovate/deep-log-inspection/blob/master/log-server/docker-compose-keystone.yml).
 
 Repository Layout
```
Comment (chicco785) [inline README.md]: use header in a constant way ###  vs ---------

File: doc/architecture.md [hunk 4]
```diff
@@ -1,13 +1,21 @@
 ## Architecture of Deep Log Inspection
 The architecture of the system can be depicted as follows:
+
 ![system architecture](img/monasca_log_api.png)
 
 The logs are sent using syslog from [FIWARE Lab][1] Nodes to the Monasca Log Agent, installed on a remote machine.
```
Comment (chicco785) [inline doc/architecture.md]: use real url [1]

File: doc/install/docker.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+## Configuring Docker for Elasticsearch
+For correctly deploying Elasticsearch, configuration of Docker must be tuned. Please refer to [Elasticsearch's official guide][1].
```
Comment (chicco785) [inline doc/install/docker.md:2]: use directly the url?

File: doc/install/install.md [hunk 6]
```diff
@@ -1,26 +1,25 @@
...
 
-Deploy the system with [Docker Swarm][3]:
+        docker stack deploy -c docker-compose.yml deeplog
 
-    docker stack deploy -c docker-compose.yml deeplog
-
-In development/test environment, you might want to start also keystone:
-
-    docker stack deploy -c docker-compose-keystone.yml deeplog
+The compose file can be found [here](https://github.com/martel-innovate/deep-log-inspection/blob/master/log-server/docker-compose.yml).
```
Comment (chicco785) [inline doc/install/install.md]: How to verify that the system is up?

curl elasticsearch health api (return the status of the cluster)

File: doc/install/traefik.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+#### Global configuration
+For convenience, most of the global configuration is managed in the [compose file](https://github.com/martel-innovate/deep-log-inspection/blob/master/log-server/docker-compose.yml), in the `command` section of `traefik`:
+
+| Setting | Description | Value |
+| --- | --- | --- |
+| `docker` | Enable Docker configuration backend | N.A. |
+| `docker.swarmmode` | Enable swarm mode on Træfik | true |
+| `docker.endpoint` | Docker server endpoint: can be a tcp or a unix socket endpoint | unix:///var/run/docker.sock |
+| `docker.domain` | Default domain used | ${DOMAIN} |
```
Comment (chicco785) [inline doc/install/traefik.md]: what is the default?

Comment (chicco785) [reply doc/install/traefik.md]: ${DOMAIN:-example.com}

File: doc/install/traefik.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+| `traefik.backend.loadbalancer.sticky` | N.A. (default: false) | false | true |
+| `traefik.backend.loadbalancer.swarm` | false | false | false |
+| `traefik.backend.circuitbreaker.expression` | N.A. | NetworkErrorRatio() > 0.5 | NetworkErrorRatio() > 0.5 |
+| `traefik.port` | 8090 | 9200 | 5601 |
+| `traefik.enable` | true | true | true |
+| `traefik.frontend.rule` | Host:${DOMAIN}; | Host:elastic.${DOMAIN} | Host:kibana.${DOMAIN} |
+|| Method:POST; |||
+|| PathPrefix:/v3.0 |||
+| `traefik.frontend.passHostHeader` | false | false | false |
+| `traefik.frontend.entryPoints` | http,https | http,https | http,https |
```
Comment (chicco785) [inline doc/install/traefik.md]: if the there is the certificate, i would keep only https

**Review by chicco785 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (k33rs) [reply README.md]: example.com, elastic.example.com and kibana.example.com are all mapped to the same IP, so I guess it's a name server with two alternative names?

File: README.md (hunk 2 above)
Comment (k33rs) [reply README.md]: it's listed, I believe it was not displayed by Github in the PR changes.

File: doc/install/docker.md (hunk 5 above)
Comment (k33rs) [reply doc/install/docker.md:2]: URL is in the footnote because it's used twice in the file.

**Comment by k33rs** [comment]
You can also see the result on [readthedocs](http://deep-log-inspection.readthedocs.io/en/docs/?badge=latest).

File: doc/architecture.md [hunk 9]
```diff
@@ -1,13 +1,19 @@
 ## Architecture of Deep Log Inspection
 The architecture of the system can be depicted as follows:
+
 ![system architecture](img/monasca_log_api.png)
 
-The logs are sent using syslog from [FIWARE Lab][1] Nodes to the Monasca Log Agent, installed on a remote machine.
+The logs are sent using syslog from [FIWARE Lab](https://www.fiware.org/lab/) Nodes to the Monasca Log Agent, installed on a remote machine.
 
-The Monasca Log Agent sends the logs to the Monasca Log API. Both the Agent and the API must be authenticated to the FIWARE Lab Keystone to function properly.
+The Monasca Log Agent sends the logs to Traefik, a HTTP reverse proxy and load balancer. Traefik then forwards the logs to the Monasca Log API. Both the Agent and the API must be authenticated to the FIWARE Lab Keystone to function properly.
```
Comment (taliaga) [inline doc/architecture.md:8]: Do we have pointers on the implications of the fact that both agent and API must be authenticated to FIWARE Lab Keystone in terms of required installation steps or considerations during configuration?

File: doc/install/install.md [hunk 10]
```diff
@@ -1,42 +1,62 @@
 ## Installing & Deploying
-Installing and running the Deep Log Inspection system requires [git][1] and [docker][2], so make sure they are installed on your system before proceeding. In order for Elasticsearch to boot correctly, user limits must be set via [Docker configuration][5].
+Installing and running the Deep Log Inspection system requires [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) and [Docker Engine](https://www.docker.com/) in [Swarm mode](https://docs.docker.com/engine/swarm/):
 
-Open your system's command line and `cd` to the folder where the project will be stored. Then, download the project by cloning the git repository:
+1. Set up the [Docker Swarm cluster](https://docs.docker.com/get-started/part4/#create-a-cluster) where services will be deployed
+2. Download the project by cloning the git repository:
```
Comment (taliaga) [inline doc/install/install.md]: I'd add "in the docker swarm master node" (same in the quickstart) :)

File: doc/install/install.md [hunk 11]
```diff
@@ -1,42 +1,62 @@
...
 
-    git clone https://github.com/martel-innovate/deep-log-inspection.git
+        git clone https://github.com/martel-innovate/deep-log-inspection.git
 
-To install and run the system, also refer to the [Quick Start guide](../quickstart.md).
+3. Set user limits in Docker [configuration file](https://github.com/martel-innovate/deep-log-inspection/blob/master/log-server/config/docker.service). For more details, refer to the [Docker configuration section](docker.md)
```
Comment (taliaga) [inline doc/install/install.md]: Again, state in which nodes this config is required.

File: doc/install/install.md [hunk 12]
```diff
@@ -1,42 +1,62 @@
...
+    * see [`preload.yml`](https://github.com/martel-innovate/deep-log-inspection/blob/master/log-server/keystone/preload.yml) for full account info
+
+To bring keystone up along with the other services, deploy it to the same stack:
 
     docker stack deploy -c docker-compose-keystone.yml deeplog
 
-Once the system is up and running, you can apply later changes to a service, e.g.:
+The compose file for keystone can be found [here](https://github.com/martel-innovate/deep-log-inspection/blob/master/log-server/docker-compose-keystone.yml).
+
+## System is up
```
Comment (taliaga) [inline doc/install/install.md]: I'd suggest removing the duplication with the quickstart until this point, feel free to get rid of the duplicated content here by adding a reference to the quickstart. You can of course add the additional info for customisation here.

File: README.md [hunk 13]
```diff
@@ -6,22 +6,36 @@ This repository contains resources for deploying the Deep Log Inspection system
 
 This project is part of [FIWARE][1]. Credits to [monasca-docker][2] and [docker-elk][3].
 
-Quick Start
------------
+## Quick Start
```
Comment (taliaga) [inline README.md:9]: Keep here just a reference to the quickstart.md to remove the duplicated content

**Review by taliaga — CHANGES_REQUESTED** [review]

File: doc/architecture.md (hunk 9 above)
Comment (k33rs) [reply doc/architecture.md:8]: Refer to installation guides for [Monasca Log API](https://github.com/martel-innovate/deep-log-inspection/blob/docs/doc/install/monasca-log-api.md) and [Monasca Log Agent](https://github.com/martel-innovate/deep-log-inspection/blob/docs/doc/install/monasca-log-agent.md).
N.B.: authentication to FIWARE Lab Keystone still has to be discussed with José.

File: README.md (hunk 13 above)
Comment (k33rs) [reply README.md:9]: done

File: doc/index.md [hunk 14]
```diff
@@ -16,8 +16,12 @@ Welcome to the documentation for Deep Log Inspection
...
-        2. [Installing & Deploying](install/install.md)
-        3. [Configuring the Monasca Log API](install/monasca-log-api.md)
-        4. [Configuring Kafka](install/kafka.md)
+        2. [Requirements](install/requirements.md)
+        3. [Installing & Deploying](install/install.md)
+        4. [Configuring Docker](install/docker.md)
+        5. [Configuring Traefik](install/traefik.md)
+        6. [Configuring the Monasca Log API](install/monasca-log-api.md)
+        7. [Configuring Kafka](install/kafka.md)
+        8. [Configuring the ELK stack](install/elk.md)
```
Comment (taliaga) [inline doc/index.md]: just a quick tip (no need to change), with longer lists (say more than 3) it's better to use always `1.` The reason being convenience when you need to edit the list.

File: doc/architecture.md (hunk 9 above)
Comment (taliaga) [reply doc/architecture.md:8]: OK, let's create an issue to keep track of this. Same for all other pending discussions we may have.

**Review by taliaga — APPROVED** [review]

**Comment by taliaga** [comment]
Much better overall :)

**Review by chicco785 — APPROVED** [review]

**Pull request merged**